src/Analyzers/CSharp/Analyzers/SimplifyPropertyAccessor/CSharpSimplifyPropertyAccessorDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,122 @@
+               new LocalizableResourceString(nameof(CSharpAnalyzersResources.Property_accessor_can_be_simplified), CSharpAnalyzersResources.ResourceManager, typeof(CSharpAnalyzersResources)))
+    {
+    }
+
+    public override DiagnosticAnalyzerCategory GetAnalyzerCategory()
+        => DiagnosticAnalyzerCategory.SyntaxTreeWithoutSemanticsAnalysis;
+
+    protected override void InitializeWorker(AnalysisContext context)
+    {
+        context.RegisterSyntaxNodeAction(AnalyzeAccessorList, SyntaxKind.AccessorList);
+    }
+
+    private void AnalyzeAccessorList(SyntaxNodeAnalysisContext context)
+    {
+        var option = context.GetCSharpAnalyzerOptions().PreferSimplePropertyAccessors;
+        if (!option.Value || ShouldSkipAnalysis(context, option.Notification))
+            return;
+
+        var accessorList = (AccessorListSyntax)context.Node;
+        var accessors = accessorList.Accessors;
+
+        if (!accessorList.Parent.IsKind(SyntaxKind.PropertyDeclaration) ||
+            !IsValidAccessorList(accessorList))
+        {
+            return;
+        }
+
+        foreach (var accessor in accessors)
+        {
+            // get { return field; }
+            // get => field;
+            if (accessor is { RawKind: (int)SyntaxKind.GetAccessorDeclaration, Body.Statements: [ReturnStatementSyntax { Expression.RawKind: (int)SyntaxKind.FieldExpression }] }
+                         or { RawKind: (int)SyntaxKind.GetAccessorDeclaration, ExpressionBody.Expression.RawKind: (int)SyntaxKind.FieldExpression })
+            {
+                ReportIfValid(accessor);
+            }
+
+            // set/init { field = value; }
+            if (accessor is { RawKind: (int)SyntaxKind.SetAccessorDeclaration or (int)SyntaxKind.InitAccessorDeclaration, Body.Statements: [ExpressionStatementSyntax { Expression: var innerBlockBodyExpression }] } &&
+                IsFieldValueAssignmentExpression(innerBlockBodyExpression))
+            {
+                ReportIfValid(accessor);
+            }
+
+            // set/init => field = value;
+            if (accessor is { RawKind: (int)SyntaxKind.SetAccessorDeclaration or (int)SyntaxKind.InitAccessorDeclaration, ExpressionBody.Expression: var innerExpressionBodyExpression } &&
+                IsFieldValueAssignmentExpression(innerExpressionBodyExpression))
+            {
+                ReportIfValid(accessor);
+            }
+        }
+
+        // One can argue that these are semantic-level checks and they would be correct.
+        // However it is very simple to perform them on our own to not fall
+        // from the quick and efficient syntax-only analyzer path
+        static bool IsValidAccessorList(AccessorListSyntax accessorList)
+        {
+            var accessors = accessorList.Accessors;
+            if (accessors.Count > 2)
```

Comment:
do you need to check you have at least one accessor?

Reply:
Not really. After this check we will enumerate all accessors and if there are none this will simply be a no-op. Added a test to verify no crashes and so on